src/SprykerEco/Yves/Heidelpay/Hydrator/EasyCreditResponseToQuoteHydrator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Yves\Heidelpay\Hydrator;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Shared\Heidelpay\HeidelpayConfig;
+use SprykerEco\Yves\Heidelpay\Handler\HeidelpayEasyCreditHandler;
+use SprykerEco\Yves\Heidelpay\Hydrator\Exception\EasyCreditResponseToQuoteHydratorException;
+
+class EasyCreditResponseToQuoteHydrator implements EasyCreditResponseToQuoteHydratorInterface
+{
+    public const PRICE_PRECISION = 100;
+
+    /**
+     * @var \SprykerEco\Yves\Heidelpay\Handler\HeidelpayEasyCreditHandler
+     */
+    private $heidelpayEasyCreditHandler;
+
+    /**
+     * @param \SprykerEco\Yves\Heidelpay\Handler\HeidelpayEasyCreditHandler $heidelpayEasyCreditHandler
+     */
+    public function __construct(HeidelpayEasyCreditHandler $heidelpayEasyCreditHandler)
+    {
+        $this->heidelpayEasyCreditHandler = $heidelpayEasyCreditHandler;
+    }
+
+    /**
+     * @param array $responseAsArray
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return void
+     */
+    public function hydrateEasyCreditResponseToQuote(array $responseAsArray, QuoteTransfer $quoteTransfer): void
+    {
+        $paymentTransfer = $quoteTransfer->requirePayment()->getPayment();
+        $paymentTransfer->setPaymentSelection(HeidelpayConfig::PAYMENT_METHOD_EASY_CREDIT);
+
+        $paymentTransfer
+            ->requireHeidelpayEasyCredit()
+            ->getHeidelpayEasyCredit()
+            ->setIdPaymentReference($responseAsArray['IDENTIFICATION.UNIQUEID'])
+            ->setAmortisationText($responseAsArray['CRITERION.EASYCREDIT_AMORTISATIONTEXT'])
+            ->setAccruingInterest(
+                $this->convertDecimalToInteger($responseAsArray['CRITERION.EASYCREDIT_ACCRUINGINTEREST'])
+            )
+            ->setTotalAmount(
+                $this->convertDecimalToInteger($responseAsArray['CRITERION.EASYCREDIT_TOTALAMOUNT'])
+            );
+
+        $quoteTransfer->setPayment($paymentTransfer);
+
+        $this->heidelpayEasyCreditHandler->addPaymentToQuote($quoteTransfer);
+    }
+
+    /**
+     * @param float $value
+     *
+     * @throws \SprykerEco\Yves\Heidelpay\Hydrator\Exception\EasyCreditResponseToQuoteHydratorException
+     *
+     * @return int
+     */
+    protected function convertDecimalToInteger(float $value): int
+    {
+        if (!is_float($value)) {
```

Comment:
You can use money plugin here.
Add it in you dependency provider:
```
protected function addMoneyPlugin(Container $container): Container
    {
        $container[static::PLUGIN_MONEY] = function () {
            return new MoneyPlugin();
        };

        return $container;
    }
```
After that you can get access to it in factory and inject it to this class.